pyigm/clustering/survey_2d2pcf.py

```diff
@@ -0,0 +1,258 @@
+      sets whether galaxy-galaxy, absorber-absorber, and galaxy-absorber analysis is on
+      So, load a good one first!
+    """
+
+    def __init__(self, field):
+        #f = copy.deepcopy(field)
+        self.fields = [field]
+
+        if field.galreal is not None:
+            self.DgDg = field.DgDg
+            self.DgRg = field.DgRg
+            self.RgRg = field.RgRg
+            self.Ngal = len(field.galreal)
+            self.gal_anly = True
+        else:
+            self.gal_anly = False
+
+        if field.absreal is not None:
+            self.DaDa = field.DaDa
+            self.DaRa = field.DaRa
+            self.RaRa = field.RaRa
+            self.Nabs = len(field.absreal)
+            self.abs_anly = True
+            # Both!
+            if field.galreal is not None:
+                self.DaDg = field.DaDg
+                self.DaRg = field.DaRg
+                self.RaDg = field.RaDg
+                self.RaRg = field.RaRg
+        else:
+            self.abs_anly = False
+
+
+        # Field edges
+        self.rbinedges = field.rbinedges # Should check that any new field has the same edges
+        self.tbinedges = field.tbinedges
+
+        # Binning
+        self.rcbins = 0.5*(self.rbinedges[:-1] + self.rbinedges[1:])
+        self.tcbins = 0.5*(self.tbinedges[:-1] + self.tbinedges[1:])
+        self.rdiff  = self.rbinedges[1:] - self.rbinedges[:-1]
+        self.tdiff  = self.tbinedges[1:] - self.tbinedges[:-1]
+
+    def addField(self, new_field):
+        """ Add a new field
+
+        Parameters
+        ----------
+        new_field : ClusterField
+
+        Returns
+        -------
+        All internal appends
+
+        """
+        # Check edges
+        if not np.isclose(new_field.rbinedges, self.rbinedges):
+            raise IOError("rbinedges of new field does not match Survey!!")
+        if not np.isclose(new_field.tbinedges, self.tbinedges):
+            raise IOError("tbinedges of new field does not match Survey!!")
+
+        #f = copy.deepcopy(field)
+        self.fields.append(new_field)
+
+        if new_field.galreal is not None:
+            self.DgDg += new_field.DgDg
+            self.DgRg += new_field.DgRg
+            self.RgRg += new_field.RgRg
+            self.Ngal += len(new_field.galreal)
+
+        if new_field.absreal is not None:
+            self.DaDa += new_field.DaDa
+            self.DaRa += new_field.DaRa
+            self.RaRa += new_field.RaRa
+            self.Nabs += len(new_field.absreal)
+            # Both?
+            if new_field.galreal is not None:
+                self.DaDg += new_field.DaDg
+                self.DaRg += new_field.DaRg
+                self.RaDg += new_field.RaDg
+                self.RaRg += new_field.RaRg
+
+    def calc_xi_gg(self, sigma=0, error=None):
+        """ Calculate xi_gg with the Landy-Szalay estimator (W3)
+
+        Parameters
+        ----------
+        sigma : float, optional
+          Smoothing of the counts
```

Comment:
Gaussian standard deviation smoothing ... defined in units of the spatial grid.

Reply:
edited